common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts

```diff
@@ -176,19 +192,23 @@ export class ApplicationTokenCredentials {
             }
         };
 
+        let msalInstance: msal.ConfidentialClientApplication;
+
         // setup msal according to parameters
         switch (this.scheme) {
             case AzureModels.Scheme.ManagedServiceIdentity:
-                this.configureMSALWithMSI(msalConfig);
+                msalInstance = this.configureMSALWithMSI(msalConfig);
```

Comment:
Small suggestion - what about `return this.configureMSALWithMSI(msalConfig);` instead of assigning to variable and break? And the same for second case.

Reply:
Generally, I prefer to avoid to return in the middle of function unless it is providing some advantages or recursive. 
I prefer to keep this in this way for future "case" conditions, instead of repeating return in every condition.